api/src/test/java/com/epam/pipeline/acl/issue/IssueApiServiceTest.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright 2017-2020 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.acl.issue;
+
+import com.epam.pipeline.controller.vo.EntityVO;
+import com.epam.pipeline.controller.vo.IssueCommentVO;
+import com.epam.pipeline.controller.vo.IssueVO;
+import com.epam.pipeline.entity.datastorage.aws.S3bucketDataStorage;
+import com.epam.pipeline.entity.issue.Issue;
+import com.epam.pipeline.entity.issue.IssueComment;
+import com.epam.pipeline.entity.security.acl.AclClass;
+import com.epam.pipeline.manager.EntityManager;
+import com.epam.pipeline.manager.issue.IssueManager;
+import com.epam.pipeline.manager.security.AuthManager;
+import com.epam.pipeline.security.acl.AclPermission;
+import com.epam.pipeline.test.acl.AbstractAclTest;
+import com.epam.pipeline.test.creator.datastorage.DatastorageCreatorUtils;
+import com.epam.pipeline.test.creator.issue.IssueCreatorUtils;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+public class IssueApiServiceTest extends AbstractAclTest {
+
+    private final EntityVO entityVO = IssueCreatorUtils.getEntityVO(ID, AclClass.DATA_STORAGE);
+    private final Issue issue = IssueCreatorUtils.getIssue(entityVO, SIMPLE_USER);
+    private final IssueVO issueVO = IssueCreatorUtils.getIssueVO(entityVO);
+    private final S3bucketDataStorage s3bucket = DatastorageCreatorUtils.getS3bucketDataStorage(ID, OWNER_USER);
```

Comment:
Let's change the filed's type and name to something more common like `AbstractSecuredEntity entity`. This way we declare that any entity can be used not only s3 storages.